openmdao/jacobians/assembled_jacobian.py

```diff
@@ -280,23 +276,16 @@ def _get_subjac_iters(self, system):
             int_mtx = self._int_mtx
             ext_mtx = self._ext_mtx[system.pathname]
             subjacs = system._subjacs_info
-            owned = system._owning_rank
-            irank = system.comm.rank
-            meta = system._var_allprocs_abs2meta
             sys_inputs = system._var_allprocs_abs2prom['input']
             sys_outputs = system._var_allprocs_abs2prom['output']
 
             if isinstance(system, Component):
                 global_conns = _empty_dict
-                iscomp = True
             else:
                 global_conns = system._conn_global_abs_in2out
-                iscomp = False
 
-            output_names = set(n for n in system._var_abs_names['output']
-                               if iscomp or owned[n] == irank or meta[n]['distributed'])
-            input_names = set(n for n in system._var_abs_names['input']
-                              if iscomp or owned[n] == irank or meta[n]['distributed'])
+            output_names = set(n for n in system._var_abs_names['output'])
```

Comment:
this can be reduced to `set(system._var_abs_names['output']) and the next line similarly.

Reply:
Fixed